balancer-js/examples/exitExactBPTIn.ts

```diff
@@ -0,0 +1,91 @@
+import dotenv from 'dotenv';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import {
+  BalancerError,
+  BalancerErrorCode,
+  BalancerSDK,
+  Network,
+  PoolModel,
+} from '../src/index';
+import { parseFixed } from '@ethersproject/bignumber';
+import { forkSetup, tokenBalances } from '../src/test/lib/utils';
+
+dotenv.config();
+
+const { ALCHEMY_URL: jsonRpcUrl } = process.env;
+
+// Slots used to set the account balance for each token through hardhat_setStorageAt
+// Info fetched using npm package slot20
+const BPT_SLOT = 0;
+
+/*
+Example showing how to use Pools module to exit pools with exact BPT in method.
+*/
+async function exitExactBPTIn() {
+  const network = Network.MAINNET;
+  const rpcUrl = 'http://127.0.0.1:8545';
+  const provider = new JsonRpcProvider(rpcUrl, network);
+  const signer = provider.getSigner();
+  const signerAddress = await signer.getAddress();
+
+  const poolId =
+    '0xa6f548df93de924d73be7d25dc02554c6bd66db500020000000000000000000e'; // 50/50 WBTC/WETH Pool
+  const bptIn = parseFixed('1', 18).toString();
+  const slippage = '200'; // 200 bps = 2%
+
+  const sdkConfig = {
+    network,
+    rpcUrl,
+  };
+  const balancer = new BalancerSDK(sdkConfig);
+
+  // Use SDK to find pool info
+  const pool: PoolModel | undefined = await balancer.poolsProvider.find(poolId);
+  if (!pool) throw new BalancerError(BalancerErrorCode.POOL_DOESNT_EXIST);
+
+  // Sets up local fork granting signer initial balances and token approvals
+  await forkSetup(
+    balancer,
+    signer,
+    [pool.address],
+    [BPT_SLOT],
+    [bptIn],
+    jsonRpcUrl as string
+  );
+
+  // Checking balances to confirm success
+  const tokenBalancesBefore = await tokenBalances(
+    balancer,
+    signer,
+    pool.tokensList
+  );
+
+  const { to, data, minAmountsOut } = await pool.buildExitExactBPTIn(
+    signerAddress,
+    bptIn,
```

Comment:
Rather than using the hardcoded value here could it be fetched with `const usersBPTBalance = await tokenBalances(balancer, signer, pool.address)` (assuming this is how you fetch the users total BPT). So that it's more clear how to get the users total BPT.

Reply:
Are you referring to the case where the user is exiting with all their BPT balance? I'm not currently handling that as a special case. The method is receiving the BPT amount as input and building the transaction with that.
If that's a common scenario that we want to handle from within the SDK, I'd create another buildExitWithAllBPT(params), which would fetch the user's BPT balance and call buildExit(params) with the fetched amount. What do you think?
Ps: I had notifications turned off and didn't see those comments until today - sorry for the late reply

Reply:
I don't think we need that special case for now.